test/Dialect/AIG/balancing.mlir

```diff
@@ -0,0 +1,34 @@
+// RUN: circt-opt %s --maximum-and-cover --aig-balance-variadic | FileCheck %s
```

Comment:
Thank you for adding test! Having a test that runs both `maximum-and-cover` and `aig-balance-variadic` is great. Can we also add a test for each of them as well?